pkg/driver/common/operator/hooks.go

```diff
@@ -75,8 +75,9 @@ func withHyperShiftNodeSelector(c *clients.Clients) (dc.DeploymentHookFunc, []fa
 	hook := func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
 		podSpec := &deployment.Spec.Template.Spec
 		// Add nodeSelector
+		genInformer, _ := c.ControlPlaneHypeInformer.ForResource(hostedControlPlaneGVR)
```

Comment:
Use a typed lister, that's why we import the generated client.
```suggestion
		hcpLister := c.ControlPlaneHypeInformer.Hypershift().V1beta1().HostedControlPlanes().Lister()
```

Reply:
`getHostedControlPlaneNodeSelector` then gets `HostedControlPlaneLister` instead of `GenericLister`, and `getHostedControlPlane` down below does not need to retype a RuntimeObject to a `HostedControlPlane`.